### ISSUE TITLE
Scene Selection

### ISSUE THREAD

**Issue by lespatots** [Foddy/node-red-contrib-huemagic#57]
Any chance this node could be enhanced like the Light and Group for 'universal mode'. I just started to try this and found I needed to create a lot of nodes and a switch to select them, whereas if I could pass in the scene name it would be much simpler?

**Comment by bakman2** [comment]
You can use the "Hue Scene" node to select scenes.

**Comment by bakman2** [comment]
I understand now, I guess it would require a new node to spit out the scene names. 

You can add a drop down to the dashboard, add the scene names and pass those to the output and connect it to a hue-scene node ?

For the dropdown options, add a string: `{"name":"<your scene name>"}` 
`label` is what you will see in the dropdown.

Use the debug node to see what is sent when you change the drop down.

**Comment by lespatots** [comment]
Yes, but that was just one use case. However, you have introduced a new thought. Perhaps there should be a capability of the various nodes “Light”, “Group” “Scene” etc to get them to output the list of options available? At present I have to set the list of choices manually but that is open to change as the bridges ‘can’ change! There are other nodes that might produce a list of options too. I have three bridges for example and if a node produced a list of these then a lot more auto set-up becomes possible? One area where everyone but Philips agree is the power-on scenario. Resetting to the last known state is a must for most people, me included. I solve this with UPS supported lights but this resolvable in a flow provided enough can be auto-discovered about the Hue environment.

Regards

**Comment by bakman2** [comment]
I am no expert in this stuff, just tried to think along with you ;)

For initial states, I use variables, which can be set and read from a file. You could write the current states and use those as last-known state. It will be quite some manual work, but it works.

**Comment by lespatots** [comment]
Yes, that is why there isn’t a general solution, or if there is it isn’t a Node-Red flow. It involves a lot of stuff if you really want to reset to any state other than ‘off’ but if reading and storing states was possible a Node-Red flow could be a community development, ‘cos there are other non-Hue devices too!

Foddy did make the changes to Light and Group for me so I’m hoping that a change to ‘Scene’ might come at some time. Information from the relevant nodes is a new suggestion but a really, good one!
